src/commands/role.ts

```diff
@@ -0,0 +1,59 @@
+import {GuildMember, Message} from 'discord.js';
+import ICommand from '../library/iCommand';
+
+// Hack for implementing with static properties/methods
+let Role: ICommand;
+export default Role = class {
+
+  /* istanbul ignore next */
+  static get description(): string {
+    return 'Assigns a role to the user on request.';
+  }
+
+  public static execute(args: string[], msg: Message) {
+    const { channel } = msg;
+    const command = args[0];
+    const roleRequest = msg.guild.roles.find(r => r.name === args[1]); // Finds role based on input
+
+    if (command === "request") {
+      if (!roleRequest) { return channel.send("Please enter a valid role. Try !role help"); } // If role is null
+      assignRoleHelper(roleRequest, msg.member, msg);
+    } else if (command === "help") {
+      getHelp(msg);
+    }
+    else {
+      return channel.send("Role command unknown. Use 'role request [rolename]' to request a role.");
+    }
+  }
+
+};
+
+const assignRoleHelper = (role: any, author: GuildMember, msg: Message) => {
+  // Constants for each possible role
+  const freshman = msg.guild.roles.find(r => r.name === "Freshman");
+  const sophomore = msg.guild.roles.find(r => r.name === "Sophomore");
+  const junior = msg.guild.roles.find(r => r.name === "Junior");
+  const senior = msg.guild.roles.find(r => r.name === "Senior");
```

Comment:
Maybe add an extra function that finds the "year roles".
You could also use destructuring to shorten the property chaining trainwreck, or just nab the roles and store it in a variable to reuse.

--- 

This is begging to be refactored.
I would aim to use an array here of the year's role names.

You would then want to find the intersection of roles & year's role names; maybe this is too crafty since there shouldn't be more than 4 years, but it would DRY up the code & helps later with validation.

I would recommend this route [[SO Link](https://stackoverflow.com/questions/1885557/simplest-code-for-array-intersection-in-javascript) - use modern way]: 
`array1.filter(value => array2.includes(value))`

Then you could loop through that collection of roles from the filter and remove them - if there was some odd case where a use had multiple roles¿